lib/util.c

```diff
@@ -369,7 +408,7 @@ doubles_almost_equal(double a, double b, double eps)
  *
  */
 static inline size_t
-positive_interval_select(double x, size_t num_intervals, double const *restrict lengths)
+positive_interval_select(double x, size_t num_intervals, const double *lengths)
```

Comment:
Why remove the ``restrict`` here?

Reply:
I was just making the parameter more readable and simple. As I was debating whether to put this `restrict` in other parameters of new function, I started learning more this keyword that is new to me. https://en.wikipedia.org/wiki/Restrict is one of the examples I was looking at to understand better.

In this particular case, given the function does not change the values of any memory locations and is always reading new memory locations. There is nothing a compiler could do to compile more optimal code given the `restrict` keyword.

I disassembled the function to confirm `restrict` has no effect (at least on x86_64). And looking at the compiled instructions and my current understanding I don't see how `restrict` could have any effect.

It's a really interesting keyword, especially since it's in C and not C++. It does seem like the same desired instructions can be generated with modern compilers by declaring and using temporary variables. I'll speculate this is part of the reason the keyword hasn't been added to C++.